src/azure-cli/azure/cli/command_modules/vm/_image_builder.py

```diff
@@ -485,7 +485,9 @@ def create_image_template(  # pylint: disable=too-many-locals, too-many-branches
                                  namespace='Microsoft.Network', type='virtualNetworks', name=vnet,
                                  child_type_1='subnets', child_name_1=subnet)
         vnet_config = VirtualNetworkConfig(subnet_id=subnet)
-    vm_profile = ImageTemplateVmProfile(vm_size=vm_size, os_disk_size_gb=os_disk_size, vnet_config=vnet_config)
+    if subnet is not None and proxy_vm_size is not None:
+        vnet_config = VirtualNetworkConfig(subnet_id=subnet, proxy_vm_size=proxy_vm_size)
```

Comment:
May I ask can `VirtualNetworkConfig` be supported only when `subnet` and `proxy_vm_size` pass in at the same time?

Reply:
If so, if users only pass in `proxy_vm_size`, do we need to consider raising an error to tell the user that `subnet` is missing?

Reply:
Yes,this is required by the service https://github.com/Azure/azure-cli/issues/20680#issue-1075955660

Reply:
OK, we can consider adding a check. When only `proxy_vm_size` is passed in, we can raising an error to prompt users `subnet` must also be passed in